kernel/Utilities/FaceLocalIndexing.h

```diff
@@ -102,6 +102,32 @@ class FaceLocalIndexing {
         }
     }
 
+    /**
+     * For a given unknown, what are the indices in the face matrx.
+     * @param unknown The unkown
+     * @param mapping Mapping from the i-th DoF/basisFunction to the
+     * corresponding row/column in the face matrix.
+     */
+    void getDoFMapping(std::size_t unknown,
+                       std::vector<std::size_t>& mapping) const {
```

Comment:
```suggestion
    std::vector<std::size_t> getDoFMapping(std::size_t unknown) const {
```

would this work?

Reply:
Could, but the primary idea here is to allow reusing the same vector without constantly having to reallocate the space.

Though to be fair, a lot of allocation is already being done.

Reply:
Yes, okay but due to the resize call it might have to reallocate anyway. So yes I see the point, but I think it is not a big optimisation. But it is a fair argument, so feel free.
If you keep it, I would add a comment.

Reply:
Yes indeed. In practice though it would grow to the size of the largest number of basis functions and then stay at that size.